src/Components/Web/src/Forms/InputCheckbox.cs

```diff
@@ -21,6 +21,11 @@ namespace Microsoft.AspNetCore.Components.Forms
     /// </summary>
     public class InputCheckbox : InputBase<bool>
     {
+        /// <summary>
+        /// Gets or sets the associated <see cref="ElementReference"/>
+        /// </summary>
+        public ElementReference? InputElement { get; protected set; }
```

Comment:
This is a slightly unusual nullable property in that it becomes available and stays non-null once the component is rendered. Would it be better to have this throw if it's accessed before it's initialized?

```C#
    return _inputElement ?? throw new InvalidOperationException($"Component must be rendered before {nameof(InputElement)} can be accessed.");

Reply:
This would also allow the property to be non-null.

Reply:
Agree